driver-core/src/main/java/com/datastax/driver/core/DataTypeParser.java

```diff
@@ -0,0 +1,255 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.ImmutableMap;
+
+import com.datastax.driver.core.exceptions.DriverInternalError;
+
+import static com.datastax.driver.core.DataType.*;
+import static com.datastax.driver.core.ParseUtils.isBlank;
+import static com.datastax.driver.core.ParseUtils.isIdentifierChar;
+import static com.datastax.driver.core.ParseUtils.skipSpaces;
+
+/*
+ * Helps transforming CQL types (as read in the schema tables) to
+ * DataTypes, starting from Cassandra 3.0.
+ *
+ * Note that these methods all throw DriverInternalError when there is a parsing
+ * problem because in theory we'll only parse class names coming from Cassandra and
+ * so there shouldn't be anything wrong with them.
+ */
+class DataTypeParser {
```

Comment:
We should rename `DataTypeParser` and `CassandraTypeParser`, the current names don't tell what's different between the two classes.
I suggest `CassandraTypeCqlNameParser` and `CassandraTypeClassNameParser`.